src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1399,7 +1518,11 @@ fn calculate_normal(
     } else {
         let dep_info = dep_info_loc(build_runner, unit);
         let dep_info = dep_info.strip_prefix(&target_root).unwrap().to_path_buf();
-        vec![LocalFingerprint::CheckDepInfo { dep_info }]
+        if build_runner.bcx.gctx.cli_unstable().checksum_freshness {
+            vec![LocalFingerprint::CheckDepInfoChecksums { dep_info }]
+        } else {
+            vec![LocalFingerprint::CheckDepInfo { dep_info }]
+        }
```

Comment:
Why did you go with separate enums?  Are we not able to read the checksum feature flag?  Would it be better to have a `checksum: bool` in it?

Reply:
Ah, I assume this is tied to the serialization.  I wonder if we should have a slight decoupling as I feel like this makes things more complicated.  I hope having two types of fingerprints is short term and so we shouldn't try to over-generalize

Reply:
Can you help be better understand what such a decoupling might look like?

Reply:
Nevermind, I think I've got it.

Reply:
Resolved in d422f64ca56a7b6dfab5c8e5d4bbb06cccbbe5b3

Reply:
The original thinking was that I should maintain backwards compatibility with prior build caches, but then I remembered that cargo releases alongside rust, and rust doesn't provide backwards compatibility of build caches.